modules/core/04-channel/keeper/keeper_test.go

```diff
@@ -544,6 +546,75 @@ func (suite *KeeperTestSuite) TestUnsetParams() {
 	})
 }
 
+func (suite *KeeperTestSuite) TestGetV2Counterparty() {
+	var path *ibctesting.Path
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"success",
+			func() {},
+			true,
+		},
+		{
+			"failure: channel not found",
+			func() {
+				path.EndpointA.ChannelID = ""
+			},
+			false,
+		},
+		{
+			"failure: channel not OPEN",
+			func() {
+				path.EndpointA.UpdateChannel(func(channel *types.Channel) { channel.State = types.TRYOPEN })
+			},
+			false,
+		},
+		{
+			"failure: channel is ORDERED",
+			func() {
+				path.EndpointA.UpdateChannel(func(channel *types.Channel) { channel.Ordering = types.ORDERED })
+			},
+			false,
+		},
+		{
+			"failure: connection not found",
+			func() {
+				path.EndpointA.UpdateChannel(func(channel *types.Channel) { channel.ConnectionHops = []string{ibctesting.InvalidID} })
+			},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		suite.Run(tc.name, func() {
+			suite.SetupTest() // reset
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			path.Setup()
+
+			tc.malleate()
+
+			counterparty, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetV2Counterparty(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+
+			if tc.expPass {
+				suite.Require().True(found)
+
+				merklePath := commitmentv2types.NewMerklePath([]byte("ibc"), []byte(""))
+				expCounterparty := packetservertypes.NewCounterparty(path.EndpointB.ClientID, path.EndpointA.ChannelID, merklePath)
```

Comment:
This is flipped. It should be:

```suggestion
				expCounterparty := packetservertypes.NewCounterparty(path.EndpointA.ClientID, path.EndpointB.ChannelID, merklePath)
```

The counterparty stores our side's clientID and the counterparty side's channel ID. I think this will be conceptually a lot lot easier to understand once we change the name of the counterparty to channel.

The test happens to still pass because this is the first client and channel on both sides so the identifiers are all the same. We should probably mess with the setup to ensure this doesn't pass if the chains are flipped